### PR TITLE
dep(*): update dependency to update dumpling and gc tuner

### PR DESCRIPTION
**Pull request by lance6716**
Signed-off-by: lance6716 <lance6716@gmail.com>

### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/7428

#### Tests 

 - Manual test [TODO]

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: engine/executor/server.go [hunk 1]
```diff
@@ -390,6 +392,15 @@ func (s *Server) isReadyToServe() bool {
 // Run drives server logic in independent background goroutines, and use error
 // group to collect errors.
 func (s *Server) Run(ctx context.Context) error {
+	if s.cfg.ExclusivelyDeployed {
+		limit, err := memory.MemTotal()
+		if err != nil {
+			log.L().Warn("get memory failed", zap.Error(err))
+		}
```
Comment (amyangfei) [inline engine/executor/server.go:399]: Is it better to reset `limit` to zero if err is not nil

File: engine/executor/server.go [hunk 2]
```diff
@@ -390,6 +392,20 @@ func (s *Server) isReadyToServe() bool {
 // Run drives server logic in independent background goroutines, and use error
 // group to collect errors.
 func (s *Server) Run(ctx context.Context) error {
+	if s.cfg.ExclusivelyDeployed {
+		limit, err := memory.MemTotal()
+		if err != nil {
+			log.L().Warn("get memory failed", zap.Error(err))
```
Comment (CharlesCheung96) [inline engine/executor/server.go:398]:
```suggestion
			log.Warn("get memory failed", zap.Error(err))
```

File: engine/executor/server.go [hunk 3]
```diff
@@ -390,6 +392,20 @@ func (s *Server) isReadyToServe() bool {
...
+			limit = 0
+		}
+		threshold := limit * 7 / 10
+		log.L().Info("set memory threshold to GC tuner",
```
Comment (CharlesCheung96) [inline engine/executor/server.go:402]:
```suggestion
		log.Info("set memory threshold to GC tuner",
```

File: deployments/engine/docker-compose/1m1e_with_s3.yaml [hunk 4]
```diff
@@ -0,0 +1,104 @@
+version: "2.3"
```
Comment (CharlesCheung96) [inline deployments/engine/docker-compose/1m1e_with_s3.yaml:1]: Why do we need this config file?

File: engine/executor/config.go [hunk 5]
```diff
@@ -144,6 +147,7 @@ func GetDefaultExecutorConfig() *Config {
 		Join:                 defaultJoinAddr,
 		Addr:                 defaultExecutorAddr,
 		AdvertiseAddr:        "",
+		ExclusivelyDeployed:  true, // currently 1 container 1 executor
```
Comment (CharlesCheung96) [inline engine/executor/config.go:150]: Is this parameter used elsewhere?

File: deployments/engine/docker-compose/1m1e_with_s3.yaml (hunk 4 above)
Comment (lance6716) [reply deployments/engine/docker-compose/1m1e_with_s3.yaml:1]: it's my local benchmark config. Maybe other can also re-use it

**Review by CharlesCheung96 — APPROVED** [review]

**Comment by lance6716** [comment]
ptal @GMHDBJD @amyangfei

File: engine/executor/config.go [hunk 6]
```diff
@@ -50,6 +50,9 @@ type Config struct {
 	AdvertiseAddr string `toml:"advertise-addr" json:"advertise-addr"`
 
 	Labels map[string]string `toml:"labels" json:"labels"`
+	// ExclusivelyDeployed means only one executor is deployed in the container/machine,
+	// which means it can consume almost all the resources of the container/machine.
+	ExclusivelyDeployed bool `toml:"exclusively-deployed" json:"exclusively-deployed"`
```
Comment (amyangfei) [inline engine/executor/config.go:55]: What about using a more explicit config, such as `EnableGCTuning`, it is difficult to determine whether enable gc tuner is safe even under exclusively deployment.

**Pull request merged**